mccodec/mc/configuration/clientbound_server_links_config.mbt

```diff
@@ -0,0 +1,100 @@
+///|
+enum ServerLinkType {
+  BugReport
+  CommunityGuidelines
+  Support
+  Status
+  Feedback
+  Community
+  Website
+  Forums
+  News
+  Announcements
+}
+
+///|
+pub fn ServerLinkType::from_int(val : Int) -> ServerLinkType {
+  match val {
+    0 => ServerLinkType::BugReport
+    1 => ServerLinkType::CommunityGuidelines
+    2 => ServerLinkType::Support
+    3 => ServerLinkType::Status
+    4 => ServerLinkType::Feedback
+    5 => ServerLinkType::Community
+    6 => ServerLinkType::Website
+    7 => ServerLinkType::Forums
+    8 => ServerLinkType::News
+    9 => ServerLinkType::Announcements
+    _ => panic()
+  }
+}
+
+///|
+pub fn ServerLinkType::to_int(self : ServerLinkType) -> Int {
+  match self {
+    ServerLinkType::BugReport => 0
+    ServerLinkType::CommunityGuidelines => 1
+    ServerLinkType::Support => 2
+    ServerLinkType::Status => 3
+    ServerLinkType::Feedback => 4
+    ServerLinkType::Community => 5
+    ServerLinkType::Website => 6
+    ServerLinkType::Forums => 7
+    ServerLinkType::News => 8
+    ServerLinkType::Announcements => 9
+  }
+}
+
+///|
+struct ServerLink {
```

Comment:
the struct definition doesn't feel intuitive and is not public(i didnt see any public constructor). Considering utilities like `Either[ServerLinkType, TextComponent]` may improve the readability much